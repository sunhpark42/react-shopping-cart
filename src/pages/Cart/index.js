import React from 'react';
import { useHistory } from 'react-router';
import { useDispatch, useSelector } from 'react-redux';
import {
  setAllCartItemCheckbox,
  toggleCartItemCheckbox,
  setCartItemQuantity,
  deleteCartItems,
} from '../../store/cartReducer';
import API from '../../request/api';
import { Button, HighlightText, NumericInput, Product, IconButton } from '../../components/shared/';
import { COLOR, MESSAGE, PATH } from '../../constants';
import {
  Header,
  Contents,
  ListOptionMenu,
  ProductListContainer,
  ProductListHeader,
  ReceiptWrapper,
  ReceiptHeader,
  ReceiptContent,
  ReceiptRow,
  CheckBox,
} from './style';
import { ReactComponent as TrashBin } from '../../assets/icons/trash-bin.svg';

const Cart = () => {
  const list = useSelector(state => state.cartReducer.cart);
  const dispatch = useDispatch();
  const history = useHistory();

  const checkedItemIdList = list.filter(item => item.checked).map(({ id }) => id);
  const checkedCount = checkedItemIdList.length;
  const isAllChecked = checkedCount && checkedCount === list.length;
  const checkOptionText = isAllChecked
    ? '선택해제'
    : checkedCount
    ? `${checkedCount}개 선택`
    : '전체선택';

  const totalPrice = list
    .filter(item => item.checked)
    .reduce((total, item) => {
      const { price, quantity } = item;
      return total + price * quantity;
    }, 0);
  const isPurchasable = totalPrice > 0;

  const onCheckBoxChange = ({ id }) => {
    dispatch(toggleCartItemCheckbox(id));
  };

  const onCheckOptionChange = () => {
    dispatch(setAllCartItemCheckbox(isAllChecked));
  };

  const onItemQuantityChange = id => quantity => {
    dispatch(setCartItemQuantity({ id, quantity }));
  };

  const onDelete = async idList => {
    if (window.confirm(MESSAGE.CONFIRM_DELETE_ITEM)) {
      try {
        await Promise.all(idList.map(id => API.deleteCartItem({ id })));
        dispatch(deleteCartItems(idList));
      } catch (error) {
        console.error(error);
        alert(MESSAGE.FAIL_DELETE_ITEM);
      }
    }
  };

  return (
    <>
      <Header>장바구니</Header>
      <Contents>
        <ProductListContainer>
          <ListOptionMenu aria-label="상품선택 옵션 메뉴">
            <CheckBox role="checkbox" aria-checked={isAllChecked}>
              <input
                type="checkbox"
                checked={isAllChecked ? true : false}
                onChange={onCheckOptionChange}
                hidden
              />
              <span></span>
              {checkOptionText}
            </CheckBox>
            <Button
              type="button"
              size="small"
              color={COLOR.BLACK}
              backgroundColor={COLOR.WHITE}
              borderColor={COLOR['GRAY-300']}
              onClick={() => onDelete(checkedItemIdList)}
              disabled={checkedCount === 0}
            >
              상품 삭제
            </Button>
          </ListOptionMenu>
          <ProductListHeader>배송상품 ({list.length}개)</ProductListHeader>
          <ul aria-label="장바구니 상품 목록">
            {list.map(({ id, name, image, price, quantity, checked }) => (
              <li key={id}>
                <CheckBox>
                  <input
                    type="checkbox"
                    onChange={() => onCheckBoxChange({ id })}
                    checked={checked}
                    hidden
                  />
                  <span role="checkbox" aria-label={`${name} 선택`} aria-checked={checked}></span>
                </CheckBox>
                <Product
                  onTitleClick={() => {
                    history.push(`${PATH.GOODS_DETAIL}/${id}`);
                  }}
                  thumbnail={{
                    image: image,
                    alt: name,
                    size: 'small',
                    onClick: () => {
                      history.push(`${PATH.GOODS_DETAIL}/${id}`);
                    },
                  }}
                  information={{ title: name }}
                  extra={
                    <>
                      <IconButton
                        type="button"
                        size="small"
                        onClick={() => onDelete([id])}
                        ariaLabel={`${name} 삭제`}
                      >
                        <TrashBin />
                      </IconButton>
                      <NumericInput
                        min={1}
                        max={99}
                        value={quantity}
                        setValue={onItemQuantityChange(id)}
                        ariaLabel={`${name} 수량 변경`}
                      />
                      <div aria-label={`${name} 합산 가격`}>
                        {`${(price * quantity).toLocaleString('ko-KR')}원`}
                      </div>
                    </>
                  }
                />
              </li>
            ))}
          </ul>
        </ProductListContainer>
        <ReceiptWrapper>
          <ReceiptHeader>결제예상금액</ReceiptHeader>
          <ReceiptContent>
            <ReceiptRow>
              <HighlightText color={COLOR.HIGHLIGHT_MINT} fontSize="1.25rem">
                결제예상금액
              </HighlightText>
              <HighlightText
                color={COLOR.HIGHLIGHT_MINT}
                fontSize="1.25rem"
                ariaLabel="결제예상금액"
              >
                {`${totalPrice.toLocaleString('ko-KR')}원`}
              </HighlightText>
            </ReceiptRow>

            <Button
              type="button"
              size="medium"
              disabled={!isPurchasable}
              onClick={() => {
                history.push(PATH.ORDER);
              }}
            >
              {`주문하기(${checkedCount}개)`}
            </Button>
          </ReceiptContent>
        </ReceiptWrapper>
      </Contents>
    </>
  );
};

export default Cart;
