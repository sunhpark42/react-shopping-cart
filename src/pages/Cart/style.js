import styled from 'styled-components';
import { COLOR } from '../../constants';

export const Header = styled.h2`
  color: ${COLOR.BLACK};
  font-size: 2rem;
  border-bottom: 4px solid ${COLOR.BLACK};
  text-align: center;
  padding-bottom: 1.5rem;
`;

export const Contents = styled.div`
  display: flex;
  padding: 52px 24px;
  justify-content: space-between;
  width: 100%;
`;

export const ProductListContainer = styled.div`
  display: flex;
  flex-direction: column;
  width: 750px;

  & li {
    display: flex;
    border-bottom: 2px solid ${COLOR['GRAY-300']};
  }
`;

export const ListOptionMenu = styled.div`
  display: flex;
  justify-content: space-between;
  margin-bottom: 32px;
`;

export const ProductListHeader = styled.h3`
  font-size: 1.25rem;
  color: ${COLOR.BLACK};
  border-bottom: 4px solid ${COLOR['GRAY-500']};
  padding-bottom: 24px;
`;

export const CheckBox = styled.label`
  display: inline-flex;
  cursor: pointer;
  align-items: center;
  position: relative;

  align-items: flex-start;
  padding-top: 1rem;

  & input:checked ~ span {
    border-color: ${COLOR.MINT};
    background-color: ${COLOR.MINT};
    color: ${COLOR.WHITE};
  }

  & > span {
    width: 28px;
    height: 28px;
    border: 1px solid ${COLOR['GRAY-100']};
    border-radius: 2px;
    transition: background-color 0.2s ease-in;
    color: transparent;
    text-align: center;
    margin-right: 0.5rem;

    &:hover {
      filter: brightness(0.95);
    }
    &:after {
      content: 'âœ“';
      font-size: 1.5rem;
    }
  }
`;

export const ReceiptWrapper = styled.div`
  width: 448px;
  max-height: 320px;
  border: 1px solid ${COLOR['GRAY-200']};
  position: sticky;
  top: 120px;
  left: 70%;
  padding-bottom: 1rem;
`;

export const ReceiptHeader = styled.h3`
  font-size: 1.5rem;
  color: ${COLOR.BLACK};
  border-bottom: 4px solid ${COLOR['GRAY-200']};
  padding: 20px 30px;
`;

export const ReceiptContent = styled.div`
  display: flex;
  height: 80%;
  flex-direction: column;
  justify-content: space-between;
  padding: 20px 30px;
`;

export const ReceiptRow = styled.div`
  display: flex;
  justify-content: space-between;
  margin-bottom: 1rem;
`;
