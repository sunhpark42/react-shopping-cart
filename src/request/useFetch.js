import { useEffect, useState } from 'react';
import { BASE_URL } from '../constants';

const useFetch = (url, options = { method: 'GET' }) => {
  const [response, setResponse] = useState([]);
  const [error, setError] = useState();

  useEffect(() => {
    const fetchData = async () => {
      try {
        const res = await fetch(`${BASE_URL}${url}`, options);

        if (!res.ok) {
          throw new Error(response.status);
        }

        if (options.method === 'GET') {
          const json = await res.json();

          setResponse(json);
        } else {
          setResponse(response);
        }
      } catch (error) {
        setError(error);
      }
    };

    fetchData();
  }, []);

  return [response, error];
};

export default useFetch;
