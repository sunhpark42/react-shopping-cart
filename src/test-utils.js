// test-utils.js
import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { combineReducers, createStore } from 'redux';
import { Provider } from 'react-redux';
// Import your own reducer

import { reducer } from './store';
import { cartReducer } from './store/cartReducer';

const rootReducer = combineReducers({ cartReducer, reducer });

function render(
  ui,
  { initialState, store = createStore(rootReducer, initialState), ...renderOptions } = {},
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from '@testing-library/react';
// override render method
export { render };
