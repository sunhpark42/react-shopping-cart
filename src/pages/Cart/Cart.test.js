import React from 'react';
import { render, fireEvent, waitFor, within, screen } from '../../test-utils';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';
import Cart from './';
import * as hooks from '../../hooks/useFetch';
import API from '../../request/api';

const initialState = {
  cartReducer: {
    cart: [],
  },
};

const cartList = [
  {
    cart_id: 1,
    product_id: 1,
    name: '[든든] 유부 슬라이스 500g',
    image_url: 'https://cdn-mart.baemin.com/goods/custom/20200525/11153-main-01.png',
    price: 4900,
  },
  {
    cart_id: 2,
    product_id: 1,
    name: '[든든] 유부 슬라이스 500g',
    image_url: 'https://cdn-mart.baemin.com/goods/custom/20200525/11153-main-01.png',
    price: 4900,
  },
  {
    cart_id: 3,
    product_id: 1,
    name: '[든든] 유부 슬라이스 500g',
    image_url: 'https://cdn-mart.baemin.com/goods/custom/20200525/11153-main-01.png',
    price: 4900,
  },
  {
    cart_id: 4,
    product_id: 1,
    name: '[든든] 유부 슬라이스 500g',
    image_url: 'https://cdn-mart.baemin.com/goods/custom/20200525/11153-main-01.png',
    price: 4900,
  },
  {
    cart_id: 5,
    product_id: 2,
    name: '[든든] 진맛살 1kg',
    image_url: 'https://cdn-mart.baemin.com/goods/custom/20200525/11157-main-01.png',
    price: 7100,
  },
];

describe('장바구니 페이지 테스트', () => {
  beforeEach(() => {
    hooks.default = jest.fn().mockReturnValue([cartList, null]);
    API.deleteCartItem = jest.fn().mockReturnValue(true);

    render(
      <Router>
        <Cart />
      </Router>,
      { initialState },
    );
  });

  it('감소 버튼을 누르면 수가 1씩 감소하고, 해당 상품의 가격과 총 금액에 1개 만큼의 금액이 감소하고, 총 금액이 변화한다.', () => {
    const number = screen.getByRole('spinbutton', {
      name: /\[든든\] 유부 슬라이스 500g 수량 변경 입력/i,
    });
    const subButton = screen.getByRole('button', {
      name: /\[든든\] 유부 슬라이스 500g 수량 변경 1개 감소/i,
    });
    const price = screen.getByLabelText('[든든] 유부 슬라이스 500g 합산 가격');
    const totalPrice = screen.getByLabelText('결제예상금액');
    fireEvent.click(subButton);
    expect(number.value).toBe('3');
    expect(price.textContent).toBe('14,700원');
    expect(totalPrice.textContent).toBe('21,800원');
  });

  it('증가 버튼을 누르면 수가 1씩 증가하고, 해당 상품의 가격과 총 금액에 1개 만큼의 금액이 증가하고, 총 금액이 변화한다.', () => {
    const number = screen.getByRole('spinbutton', {
      name: /\[든든\] 유부 슬라이스 500g 수량 변경 입력/i,
    });
    const addButton = screen.getByRole('button', {
      name: /\[든든\] 유부 슬라이스 500g 수량 변경 1개 증가/i,
    });
    const price = screen.getByLabelText('[든든] 유부 슬라이스 500g 합산 가격');
    const totalPrice = screen.getByLabelText('결제예상금액');
    fireEvent.click(addButton);
    expect(number.value).toBe('5');
    expect(price.textContent).toBe('24,500원');
    expect(totalPrice.textContent).toBe('31,600원');
  });

  it('수량 입력 칸에 20을 입력할 시, 수량이 20이 되고, 해당 상품 가격과 총 금액이 변화한다.', () => {
    const number = screen.getByRole('spinbutton', {
      name: /\[든든\] 유부 슬라이스 500g 수량 변경 입력/i,
    });
    const price = screen.getByLabelText('[든든] 유부 슬라이스 500g 합산 가격');
    const totalPrice = screen.getByLabelText('결제예상금액');
    fireEvent.change(number, { target: { value: 20 } });
    expect(number.value).toBe('20');
    expect(price.textContent).toBe('98,000원');
    expect(totalPrice.textContent).toBe('105,100원');
  });

  it('장바구니 삭제 버튼 클릭 시, 해당 아이템이 삭제된다.', async () => {
    const deleteButton = screen.getByRole('button', {
      name: /\[든든\] 유부 슬라이스 500g 삭제/i,
    });
    const totalPrice = screen.getByLabelText('결제예상금액');
    const confirmSpy = jest.spyOn(window, 'confirm');

    confirmSpy.mockImplementation(jest.fn(() => true));
    expect(totalPrice.textContent).toBe('26,700원');
    fireEvent.click(deleteButton);
    expect(confirmSpy).toBeCalled();
    await waitFor(() => expect(totalPrice.textContent).toBe('7,100원'));
  });

  it('장바구니 상품에 있는 모든 상품의 checked 상태가 true일 때 선택 해제 라벨이 보여진다.', () => {
    const listOptionMenu = screen.getByLabelText('상품선택 옵션 메뉴');
    const checkOptionLabel = within(listOptionMenu).getByRole('checkbox');
    expect(checkOptionLabel.textContent).toBe('선택해제');
    expect(checkOptionLabel).toBeChecked();
  });

  it('선택 해제 라벨을 클릭 시, 라벨 이름 전체선택으로 변경, 모든 상품 checked 해제, 결제 예상금액 0원으로 변경된다.', () => {
    const listOptionMenu = screen.getByLabelText('상품선택 옵션 메뉴');
    const checkOptionLabel = within(listOptionMenu).getByRole('checkbox');
    const totalPrice = screen.getByLabelText('결제예상금액');
    const productList = screen.getByLabelText('장바구니 상품 목록');
    const checkBoxes = within(productList).getAllByRole('checkbox');

    fireEvent.click(checkOptionLabel);
    expect(checkOptionLabel.textContent).toBe('전체선택');
    expect(checkOptionLabel).not.toBeChecked();
    checkBoxes.forEach(checkbox => expect(checkbox).not.toBeChecked());
    expect(totalPrice.textContent).toBe('0원');
  });

  it('전체 선택 라벨을 클릭 시, 라벨 이름 선택해제로 변경, 모든 상품 checked 설정, 결제 예상금액이 알맞게 변경된다.', () => {
    const listOptionMenu = screen.getByLabelText('상품선택 옵션 메뉴');
    const checkOptionLabel = within(listOptionMenu).getByRole('checkbox');
    const totalPrice = screen.getByLabelText('결제예상금액');
    const productList = screen.getByLabelText('장바구니 상품 목록');
    const checkBoxes = within(productList).getAllByRole('checkbox');

    fireEvent.click(checkOptionLabel);
    fireEvent.click(checkOptionLabel);

    expect(checkOptionLabel.textContent).toBe('선택해제');
    expect(checkOptionLabel).toBeChecked();

    checkBoxes.forEach(checkbox => expect(checkbox).toBeChecked());

    expect(totalPrice.textContent).toBe('26,700원');
  });

  it('모든 상품이 선택해제되어 있을 때 상품삭제버튼과, 결제버튼이 비활성화 된다.', () => {
    const listOptionMenu = screen.getByLabelText('상품선택 옵션 메뉴');
    const checkOptionLabel = within(listOptionMenu).getByRole('checkbox');
    fireEvent.click(checkOptionLabel);

    const deleteButton = screen.getByRole('button', { name: /상품 삭제/i });
    const purchaseButton = screen.getByRole('button', { name: /주문하기\(0개\)/i });

    expect(deleteButton).toBeDisabled();
    expect(purchaseButton).toBeDisabled();
  });

  it('상품이 2개이상일 때 1개만 선택 되어 있으면 1개 선택 라벨이 보여진다.', () => {
    const listOptionMenu = screen.getByLabelText('상품선택 옵션 메뉴');
    const checkOptionLabel = within(listOptionMenu).getByRole('checkbox');

    fireEvent.click(checkOptionLabel);

    const productList = screen.getByLabelText('장바구니 상품 목록');
    const checkBoxes = within(productList).getAllByRole('checkbox');

    fireEvent.click(checkBoxes[0]);

    expect(checkOptionLabel.textContent).toBe('1개 선택');
    expect(checkOptionLabel).not.toBeChecked();
  });

  it('상품이 2개일 때, 1개가 선택되어있을 시 1개 선택 라벨을 누르면 모든 상품의 checked 상태가 true가 된다.', () => {
    const listOptionMenu = screen.getByLabelText('상품선택 옵션 메뉴');
    const checkOptionLabel = within(listOptionMenu).getByRole('checkbox');

    fireEvent.click(checkOptionLabel);

    const productList = screen.getByLabelText('장바구니 상품 목록');
    const checkBoxes = within(productList).getAllByRole('checkbox');

    fireEvent.click(checkBoxes[0]);

    expect(checkOptionLabel.textContent).toBe('1개 선택');
    expect(checkOptionLabel).not.toBeChecked();

    fireEvent.click(checkOptionLabel);

    expect(checkOptionLabel.textContent).toBe('선택해제');
    expect(checkOptionLabel).toBeChecked();
    checkBoxes.forEach(checkbox => expect(checkbox).toBeChecked());
  });

  it('상품 삭제 버튼 클릭 시, 선택되어 있는 아이템이 삭제된다.', async () => {
    const deleteButton = screen.getByRole('button', { name: /상품 삭제/i });
    const productList = screen.getByLabelText('장바구니 상품 목록');
    const [targetItemCheckbox] = within(productList).getAllByRole('checkbox');
    const productNames = within(productList)
      .getAllByRole('heading')
      .map(heading => heading.textContent);
    const [productListHeading, targetProductName] = productNames;
    const confirmSpy = jest.spyOn(window, 'confirm');
    confirmSpy.mockImplementation(jest.fn(() => true));

    fireEvent.click(targetItemCheckbox);
    fireEvent.click(deleteButton);
    expect(confirmSpy).toBeCalled();

    await waitFor(() => {
      const names = within(productList)
        .getAllByRole('heading')
        .map(heading => heading.textContent);
      expect(names.includes(targetProductName)).toBe(false);
    });
  });

  it('결제 버튼 클릭 시 주문/결제 페이지로 이동한다.', () => {
    const purchaseButton = screen.getByRole('button', { name: /주문하기\(2개\)/i });
    fireEvent.click(purchaseButton);
    expect(window.location.href).toBe('http://localhost/order');
  });
});
