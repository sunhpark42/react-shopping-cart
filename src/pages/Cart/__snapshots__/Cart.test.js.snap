// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cart /> matches snapshot 1`] = `
<div>
  <div
    class="sc-ieebsP cLKEWA"
  >
    <h2
      class="sc-dJjZJu fqvgKC"
    >
      장바구니
    </h2>
    <div
      class="sc-hGPAah jgGCxB"
    >
      <div
        class="sc-dlVyqM bLAYPQ"
      >
        <div
          aria-label="상품선택 옵션 메뉴"
          class="sc-kfPtPH lasNxH"
        >
          <label
            aria-checked="true"
            class="sc-llYToB jSdIpu"
            role="checkbox"
          >
            <input
              checked=""
              hidden=""
              type="checkbox"
            />
            <span />
            선택해제
          </label>
          <button
            class="sc-furvIG fCAsIi"
            color="#333333"
            type="button"
          >
            상품 삭제
          </button>
        </div>
        <div
          aria-label="장바구니 상품 목록"
          class="sc-fKVsgm"
        >
          <h3
            class="sc-bBHwJV jWujwF"
          >
            배송상품 (
            2
            개)
          </h3>
          <div
            class="sc-iwjezw"
          >
            <div
              class="sc-cxpRKc krDmGH"
            >
              <label
                class="sc-llYToB jSdIpu"
              >
                <input
                  checked=""
                  hidden=""
                  type="checkbox"
                />
                <span
                  aria-checked="true"
                  aria-label="[든든] 유부 슬라이스 500g 선택"
                  role="checkbox"
                />
              </label>
              <div
                class="sc-bqiQRQ eEanMH"
              >
                <div
                  class="sc-gsDJrp"
                >
                  <div
                    class="sc-dkPtyc jnrmNB"
                  >
                    <img
                      alt="[든든] 유부 슬라이스 500g"
                      class="sc-hKwCoD doJTuW"
                      src="https://cdn-mart.baemin.com/goods/custom/20200525/11153-main-01.png"
                    />
                  </div>
                </div>
                <div
                  class="sc-ksdxAp jfWVEg"
                >
                  <h4
                    class="sc-hBURRC dxyBOS"
                  >
                    [든든] 유부 슬라이스 500g
                  </h4>
                  <div
                    class="sc-fotPbf jqZEHt"
                  />
                </div>
                <div
                  class="sc-fFehDp bItAXM"
                >
                  <button
                    aria-label="[든든] 유부 슬라이스 500g 삭제"
                    class="sc-bkkfTU ixEQNr"
                    type="button"
                  >
                    <svg>
                      trash-bin.svg
                    </svg>
                  </button>
                  <div
                    class="sc-jrQzUz fkRfCi"
                  >
                    <input
                      aria-label="[든든] 유부 슬라이스 500g 수량 변경 입력"
                      class="sc-kDThTU eoA-Dnz"
                      max="99"
                      min="1"
                      type="number"
                      value="4"
                    />
                    <div
                      class="sc-iqsfdx fEQBmJ"
                    >
                      <button
                        aria-label="[든든] 유부 슬라이스 500g 수량 변경 1개 증가"
                        class="sc-crHlIS kvGdlv"
                      >
                        <img
                          alt="증가"
                          class="sc-egiSv jRJOfA"
                          src="arrow-up.svg"
                        />
                      </button>
                      <button
                        class="sc-crHlIS kvGdlv"
                      >
                        <img
                          alt="감소"
                          aria-label="[든든] 유부 슬라이스 500g 수량 변경 1개 감소"
                          class="sc-egiSv jRJOfA"
                          src="arrow-down.svg"
                        />
                      </button>
                    </div>
                  </div>
                  <div
                    aria-label="[든든] 유부 슬라이스 500g 합산 가격"
                  >
                    19,600원
                  </div>
                </div>
              </div>
            </div>
            <div
              class="sc-cxpRKc krDmGH"
            >
              <label
                class="sc-llYToB jSdIpu"
              >
                <input
                  checked=""
                  hidden=""
                  type="checkbox"
                />
                <span
                  aria-checked="true"
                  aria-label="[든든] 진맛살 1kg 선택"
                  role="checkbox"
                />
              </label>
              <div
                class="sc-bqiQRQ eEanMH"
              >
                <div
                  class="sc-gsDJrp"
                >
                  <div
                    class="sc-dkPtyc jnrmNB"
                  >
                    <img
                      alt="[든든] 진맛살 1kg"
                      class="sc-hKwCoD doJTuW"
                      src="https://cdn-mart.baemin.com/goods/custom/20200525/11157-main-01.png"
                    />
                  </div>
                </div>
                <div
                  class="sc-ksdxAp jfWVEg"
                >
                  <h4
                    class="sc-hBURRC dxyBOS"
                  >
                    [든든] 진맛살 1kg
                  </h4>
                  <div
                    class="sc-fotPbf jqZEHt"
                  />
                </div>
                <div
                  class="sc-fFehDp bItAXM"
                >
                  <button
                    aria-label="[든든] 진맛살 1kg 삭제"
                    class="sc-bkkfTU ixEQNr"
                    type="button"
                  >
                    <svg>
                      trash-bin.svg
                    </svg>
                  </button>
                  <div
                    class="sc-jrQzUz fkRfCi"
                  >
                    <input
                      aria-label="[든든] 진맛살 1kg 수량 변경 입력"
                      class="sc-kDThTU eoA-Dnz"
                      max="99"
                      min="1"
                      type="number"
                      value="1"
                    />
                    <div
                      class="sc-iqsfdx fEQBmJ"
                    >
                      <button
                        aria-label="[든든] 진맛살 1kg 수량 변경 1개 증가"
                        class="sc-crHlIS kvGdlv"
                      >
                        <img
                          alt="증가"
                          class="sc-egiSv jRJOfA"
                          src="arrow-up.svg"
                        />
                      </button>
                      <button
                        class="sc-crHlIS kvGdlv"
                      >
                        <img
                          alt="감소"
                          aria-label="[든든] 진맛살 1kg 수량 변경 1개 감소"
                          class="sc-egiSv jRJOfA"
                          src="arrow-down.svg"
                        />
                      </button>
                    </div>
                  </div>
                  <div
                    aria-label="[든든] 진맛살 1kg 합산 가격"
                  >
                    7,100원
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-iJKOzS cRmxz"
      >
        <h3
          class="sc-giYgFv kelwgJ"
        >
          결제예상금액
        </h3>
        <div
          class="sc-ezbkgU pdkco"
        >
          <div
            class="sc-bYoCmx imTysK"
          >
            <span
              class="sc-pVTma ixgaok"
              color="#94E0DD"
              font-size="1.25rem"
            >
              결제예상금액
            </span>
            <span
              aria-label="결제예상금액"
              class="sc-pVTma ixgaok"
              color="#94E0DD"
              font-size="1.25rem"
            >
              26,700원
            </span>
          </div>
          <button
            class="sc-furvIG fkHJhn"
            color="#FFFFFF"
            type="button"
          >
            주문하기(2개)
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
